spec/rspec/core/formatters/exception_presenter_spec.rb

```diff
@@ -202,6 +202,26 @@ def initialize(message, backtrace = [], cause = nil)
         EOS
       end
 
+      it 'wont fail for the exception with a nil backtrace', :if => RSpec::Support::RubyFeatures.supports_exception_cause? do
+        allow(first_exception).to receive(:backtrace) { nil }
```

Comment:
Can you find a real scenario where this would occur? We don't tend to defensively program around "maybes", we would want an actual scenario where this can occur.

Reply:
```ruby
begin
  raise StandardError, 'oops', nil
rescue => e
  puts "backtrace: #{e.backtrace}"
end
```
fails for me with:
```
backtrace: ["1.rb:2:in `<main>'"]
```
Maybe a custom exception would do?

Reply:
It won't work with the `StandardError` because the `StandardError` doesn't have the `cause`:

```ruby
StandardError.new('foo').cause
# => nil
```

But it does fails for me in the real test when a custom error having the `#cause` is raised, but the last error in the chain doesn't have a backtrace set.

Reply:
The best is to use such a custom exception in a scenario, then.

Reply:
Sure! I've updated the test: https://github.com/rspec/rspec-core/pull/2903/files#diff-829be18287a3c0ec09a70d0505b7cc835c298063c15edce2e5a4bf20677e42f2R205

I realized that we don't have a scenario for the `[]` case too. So I've added both scenarios.
Does it make more sense now?

Reply:
Hey @JonRowe @pirj! Please let me know if the proposed changes are fine or should I improve something else here :smile: 
Also some of the environment specs on CI are failling. That's what I see in the output:
```
script/functions.sh: line 42: bin/rspec: No such file or directory
Error: Process completed with exit code 127.
```
Do you think it is fine or is it something that I might have touched with that change?

Reply:
I suggest you to rebase the PR, the error should disappear.

Will take a look a bit later. Thanks for pushing this forward